### PR TITLE
Removing IsSIMDVectorAssembly and obsoleting isInSIMDModule

### PR DESCRIPTION
**Pull request by tannergooding**
This resolves https://github.com/dotnet/coreclr/issues/27402.

Now that all of the types from `System.Numerics.Vectors.dll` have been moved down to corelib, we can remove the logic that was supporting the SIMD intrinsic types in other assemblies.

### CODE REVIEW

**Comment by tannergooding** [comment]
CC. @jkotas, @caroleidt

File: src/jit/compiler.h [hunk 1]
```diff
@@ -7770,7 +7770,13 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 
     bool isSIMDClass(CORINFO_CLASS_HANDLE clsHnd)
     {
-        return info.compCompHnd->isInSIMDModule(clsHnd);
+        if (isIntrinsicType(clsHnd))
```
Comment (jkotas) [inline src/jit/compiler.h]: It is unfortunate that `isIntrinsicType` calls `getClassAttribs`. `getClassAttribs` does a lot more. Would it make sense to rename `isInSIMDModule` on JIT/EE interface to `isIntrinsicType` ?

Comment (tannergooding) [reply src/jit/compiler.h]: I'd be fine with doing so, I'll update after I get back home tonight :)

Comment (tannergooding) [reply src/jit/compiler.h]: Done. I think this makes the overall logic quite a bit simpler. We were already caching `isIntrinsicType` in the VM, so this reduces the cost of checking elsewhere.

**Comment by tannergooding** [comment]
Renamed `isInSIMDModule` to `isIntrinsicType`, changed the implementation to returned the cached `isIntrinsicType` value stored in the method table, updated relevant callsites in the interpreter/compiler to use the new method, and updated the JIT/EE Interface Version.

File: src/ToolBox/superpmi/superpmi-shared/methodcontext.h [hunk 2]
```diff
@@ -1488,7 +1488,7 @@ enum mcPackets
     Packet_IsCompatibleDelegate                          = 99,
     Packet_IsDelegateCreationAllowed                     = 155,
     Packet_IsFieldStatic                                 = 137, // Added 4/9/2013 - needed for 4.5.1
-    Packet_IsInSIMDModule                                = 148, // Added 6/18/2014 - SIMD support
+    Packet_IsIntrinsicType                               = 148, // Added 10/26/2019 - SIMD support
```
Comment (tannergooding) [inline src/ToolBox/superpmi/superpmi-shared/methodcontext.h]: Does the value tied to the packet name need to differ?

File: src/jit/compiler.h [hunk 3]
```diff
@@ -7770,12 +7770,18 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
...
+            const char* namespaceName = nullptr;
+            (void)getClassNameFromMetadata(clsHnd, &namespaceName);
+            return strcmp(namespaceName, "System.Numerics") == 0;
+        }
+        return false;
     }
 
     bool isIntrinsicType(CORINFO_CLASS_HANDLE clsHnd)
     {
-        return (info.compCompHnd->getClassAttribs(clsHnd) & CORINFO_FLG_INTRINSIC_TYPE) != 0;
```
Comment (tannergooding) [inline src/jit/compiler.h]: I grepped for other usages of `CORINFO_FLG_INTRINSIC_TYPE` and confirmed this was the only usage in the compiler.

**Comment by tannergooding** [comment]
Updated crossgen2, which also defines `JITEEVersionIdentifier`. The two need to stay in sync and be updated simultaneously.

**Comment by CarolEidt** [comment]
Nice cleanup - thanks @tannergooding !

**Pull request merged**